src/main/java/org/terasology/rendering/nui/layers/hud/InventoryHud.java

```diff
@@ -69,4 +80,28 @@ public Boolean get() {
             return component != null && component.slot == slot;
         }
     }
+
+    private class TargetSlotBinding implements Binding<Integer> {
+
+        private int offset;
+        private LocalPlayer localPlayer;
+
+        public TargetSlotBinding(int targetSlot, LocalPlayer localPlayer) {
+            this.offset = targetSlot;
+            this.localPlayer = localPlayer;
+        }
+
+        @Override
+        public Integer get() {
+            SelectedInventorySlotComponent component =
+                    localPlayer.getCharacterEntity().getComponent(SelectedInventorySlotComponent.class);
+            return (component.slot + offset) % 10;
+            //return null;
```

Comment:
is that still required? if not please remove

Reply:
`TargetSlotBinding` gives an error if it has no getter, but are you referring to the code inside the getter instead of the getter itself? If yes, I should look more into it because that code was suggested by skaldarnar, who looked into the binding.

Reply:
Okay after looking a bit into this, I believe it is needed because one might not need the new kind of quickslot mode. I did try to remove parts I thought might be unuseful, but kept getting an error message each time. I shall tag @skaldarnar to give a second opinion.